pkg/maps/egressmap/policy.go

```diff
@@ -66,12 +68,29 @@ type policyMap struct {
 	m *ebpf.Map
 }
 
-func createPolicyMapFromDaemonConfig(daemonConfig *option.DaemonConfig, lc hive.Lifecycle, cfg PolicyConfig) bpf.MapOut[PolicyMap] {
-	if !daemonConfig.EnableIPv4EgressGateway {
-		return bpf.NewMapOut[PolicyMap](nil)
+func createPolicyMapFromDaemonConfig(in struct {
```

Comment:
nit: I don't love the use of these anonymous structs, wouldn't breaking them out make the code more readable?

Reply:
Not sure I love them either. Really for this function I'd just like to take / return three arguments, but I can't since hive needs the In and out annotations. Is it better to come up with a type name for each of these? `type policyParams` is ok, `type policyOut` is kinda wonky imo.

How would named types make this more readable? Because the function signature would be more conventional?